content/3_Silver/Prefix_Sums.mdx

```diff
@@ -681,6 +690,371 @@ as expected.
 
 Since no matter the size of the submatrix we are summing, we only need to access four values of the 2D prefix sum array, this runs in $\mathcal{O}(1)$ per query after an $\mathcal{O}(NM)$ preprocessing.
 
+## Creation From Endpoints
+
+<FocusProblem problem={problems.sample3[0]} />
+
+### 1-Dimensional
+
+A common application of prefix sums is to determine, given a set of intervals, the number of intervals that cover any particular point.
+
+In the focus problem, we are given $n$ intervals $[a,b]$, and we want to find the maximum number of intervals that cover any particular point.
+
+An naïve approach to this problem is to compute the intervals that cover each point by looping through each index from $[a,b]$ individually and incrementing by 1, resulting in an $\mathcal{O}(NQ)$ time complexity (where N is the maximum size of our interval and Q is the number of queries), which easily times out (think what happens when the interval [0, N] is queried many times).
+
+We can do better. It's easy to see that an increment of $x$ (before computation) in $\texttt{arr}[i]$ causes all subsequent $\texttt{prefix}[i...N]$ (after computation) to increase by $x$. This concept can be conceptualized through increment and decrement points. An increment point increases (and decrement decreases) all subsequent cells. (Note that our decrement point is located at $B+1$ because the interval is inclusive.)
+
+Example: Add two to each point in the interval [2, 4]
+
+Our matrix after adding 2 at our increment (start) point (before computation)
+<center>
+	<table className='text-center'>
+		<tr>
+			<td>0</td>
+			<td>0</td>
+			<td className='bg-green-200'>2</td>
```

Comment:
Normally I use light mode but I happened to be on dark mode today and this looks atrocious lmao.

@thecodingwizard is there an easy fix?

Reply:
oh yea it is lmao. try adding the dark class elements to the tags as well and it should fix. used the dark colors used on previous sections of this they look pretty good in dark mode.

Reply:
LOL when I previewed it on my end I forgot I had my dark reader on so it looked fine.

Reply:
dark:bg-red-800 or whatever. Just prefix with dark: